### PR TITLE
python311Packages.ratarmountcore: 0.6.0 -> 0.6.3

### LINKED ISSUE #293196: Build failure: python3Packages.ratarmountcore
### Steps To Reproduce

1. build *python3Packages.ratarmountcore*

### Build log

https://hydra.nixos.org/build/251414453/nixlog/1

```
Running phase: pytestCheckPhase
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.11.8, pytest-7.4.4, pluggy-1.4.0
rootdir: /build/ratarmount-ea43572/core
collected 349 items                                                            

tests/test_AutoMountLayer.py .........                                   [  2%]
tests/test_BlockParallelReaders.py ..................................... [ 13%]
........................................................................ [ 33%]
........................................................................ [ 54%]
........................................................................ [ 75%]
.............                                                            [ 78%]
tests/test_SQLiteBlobFile.py ..                                          [ 79%]
tests/test_SQLiteIndex.py .                                              [ 79%]
tests/test_SQLiteIndexedTar.py ..............................            [ 88%]
tests/test_StenciledFile.py .......................                      [ 94%]
tests/test_UnionMountSource.py ....                                      [ 95%]
tests/test_compressions.py ....                                          [ 97%]
tests/test_factory.py ...                                                [ 97%]
tests/test_utils.py .......                                              [100%]
0 assertions tested.

======================= 349 passed in 149.16s (0:02:29) ========================
Detected Python finalization from running rapidgzip thread.
To avoid this exception you should close all RapidgzipFile objects correctly,
or better, use the with-statement if possible to automatically close it.
terminate called without an active exception
/nix/store/rahjalm3cqgw5vzn6vvvj3ciw2vi4ica-pytest-check-hook/nix-support/setup-hook: line 53:   291 Aborted                 (core dumped) /nix/store/sxr2igfkwhxbagri49b8krmcqz168sim-python3-3.11.8/bin/python3.11 -m pytest
/nix/store/3qnm3nwjajgqa771dmi2dnwxrw0kzq5m-stdenv-linux/setup: line 1578: pop_var_context: head of shell_variables not a function context
```

### Additional context

This first occurs after updates to `indexed-gzip` and `ratarmountcore` itself.
The failure is not actually in a test, and occurs after the tests run. This indicates that some difference in the way the derivation is executing tests compared to upstream's CI is causing the error.

### Notify maintainers

@mxmlnkn

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Not relevant; the log is on Hydra.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by mxmlnkn**
## Description of changes
fixes https://github.com/NixOS/nixpkgs/issues/293196

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/ratarmountcore/default.nix [hunk 1]
```diff
@@ -15,19 +15,15 @@
 
 buildPythonPackage rec {
   pname = "ratarmountcore";
-  version = "0.6.0";
+  version = "0.6.3";
   format = "setuptools";
```
Comment (dotlambda) [inline pkgs/development/python-modules/ratarmountcore/default.nix:19]:
```suggestion
  pyproject = true;
```
and add setuptools to nativeBuildInputs.

File: pkgs/development/python-modules/ratarmountcore/default.nix [hunk 2]
```diff
@@ -15,19 +15,15 @@
...
 
   disabled = pythonOlder "3.6";
 
   src = fetchgit {
```
Comment (dotlambda) [inline pkgs/development/python-modules/ratarmountcore/default.nix:23]:
```suggestion
  src = fetchFromGitHub {
```

**Review by dotlambda — COMMENTED** [review]
`meta.platforms` can be removed

`passthru.optional-dependencies` would be nice

please update the commit message

**Comment by mxmlnkn** [comment]
@dotlambda 

> `meta.platforms` can be removed
> 
> `passthru.optional-dependencies` would be nice
> 
> please update the commit message

What should I use for the commit message? It is the same message one the auto-update bot uses.

After the update to 0.6.3, some of the tests suddenly do not run correctly anymore. I get this error:

```
============================= test session starts ==============================
platform linux -- Python 3.11.8, pytest-7.4.4, pluggy-1.4.0
rootdir: /tmp/nix-build-python3.11-ratarmountcore-0.6.3.drv-0/ratarmount/core
collected 353 items                                                            

tests/test_AutoMountLayer.py .........                                   [  2%]
tests/test_BlockParallelReaders.py ...FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FF [ 13%]
F.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FF [ 33%]
F.FFF.FFF.FFF.FFF.FFF.FFF.....FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FF [ 53%]
F.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FFF.FF [ 74%]
F.FFF.FFF.FFF                                                            [ 77%]
tests/test_SQLiteBlobFile.py ..                                          [ 78%]
tests/test_SQLiteIndex.py .                                              [ 78%]
tests/test_SQLiteIndexedTar.py ..............................            [ 87%]
tests/test_StenciledFile.py .......................                      [ 93%]
tests/test_SubvolumesMountSource.py ....                                 [ 94%]
tests/test_UnionMountSource.py ....                                      [ 96%]
tests/test_compressions.py ....                                          [ 97%]
tests/test_factory.py ...                                                [ 98%]
tests/test_utils.py .......                                              [100%]

=================================== FAILURES ===================================
______________________ TestParallelXZReader.test_empty[2] ______________________

________ TestParallelZstdReader.test_small_multi_stream_block[7-7-1-3] _________

self = <test_BlockParallelReaders.TestParallelZstdReader object at 0x7f57fc6121d0>
parallelization = 3, streams = 1, blocksPerStream = 7, blockSize = 7
tmp_path = PosixPath('/tmp/nix-build-python3.11-ratarmountcore-0.6.3.drv-0/pytest-of-nixbld1/pytest-0/test_small_multi_stream_block_206')

    @pytest.mark.parametrize("streams", [1, 2, 7])
    @pytest.mark.parametrize("blocksPerStream", [1, 2, 7])
    @pytest.mark.parametrize("blockSize", [1, 2, 7])
    def test_small_multi_stream_block(self, parallelization, streams, blocksPerStream, blockSize, tmp_path):
        archivePath = os.path.join(str(tmp_path), 'test-archive.zst')
        self._createArchive(archivePath, streams, blocksPerStream, blockSize)
        for bufferSize in [1, 2, 100, 1024]:
>           self._testSequentialReading(archivePath, bufferSize, parallelization)

tests/test_BlockParallelReaders.py:282: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_BlockParallelReaders.py:225: in _testSequentialReading
    parallelData = parallelFile.read(bufferSize)
ratarmountcore/BlockParallelReaders.py:328: in read
    return super()._read(size, ParallelZstdReader._decodeBlock)
ratarmountcore/BlockParallelReaders.py:176: in _read
    pool = self._getPool()
ratarmountcore/BlockParallelReaders.py:81: in _getPool
    self._pool = multiprocessing.pool.Pool(self.parallelization, self.initWorker, self.initArgs)
/nix/store/sxr2igfkwhxbagri49b8krmcqz168sim-python3-3.11.8/lib/python3.11/multiprocessing/pool.py:215: in __init__
    self._repopulate_pool()
/nix/store/sxr2igfkwhxbagri49b8krmcqz168sim-python3-3.11.8/lib/python3.11/multiprocessing/pool.py:306: in _repopulate_pool
    return self._repopulate_pool_static(self._ctx, self.Process,
/nix/store/sxr2igfkwhxbagri49b8krmcqz168sim-python3-3.11.8/lib/python3.11/multiprocessing/pool.py:329: in _repopulate_pool_static
    w.start()
/nix/store/sxr2igfkwhxbagri49b8krmcqz168sim-python3-3.11.8/lib/python3.11/multiprocessing/process.py:121: in start
    self._popen = self._Popen(self)
/nix/store/sxr2igfkwhxbagri49b8krmcqz168sim-python3-3.11.8/lib/python3.11/multiprocessing/context.py:281: in _Popen
    return Popen(process_obj)
/nix/store/sxr2igfkwhxbagri49b8krmcqz168sim-python3-3.11.8/lib/python3.11/multiprocessing/popen_fork.py:19: in __init__
    self._launch(process_obj)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <multiprocessing.popen_fork.Popen object at 0x7f57faca4190>
process_obj = <ForkProcess name='ForkPoolWorker-173' parent=11480 initial daemon>

    def _launch(self, process_obj):
        code = 1
        parent_r, child_w = os.pipe()
        child_r, parent_w = os.pipe()
>       self.pid = os.fork()
E       OSError: [Errno 12] Cannot allocate memory

/nix/store/sxr2igfkwhxbagri49b8krmcqz168sim-python3-3.11.8/lib/python3.11/multiprocessing/popen_fork.py:66: OSError
----------------------------- Captured stderr call -----------------------------
/tmp/nix-build-python3.11-ratarmountcore-0.6.3.drv-0/pytest-of-nixbld1/pytest-0/test_small_multi_stream_block_206/test-archive.zst.tmp :126.53%   (    49 B =>     62 B, /tmp/nix-build-python3.11-ratarmountcore-0.6.3.drv-0/pytest-of-nixbld1/pytest-0/test_small_multi_stream_block_206/test-archive.zst.tmp.zst) 
```

I searched around, and my guess is that for some reason the process has unusually large virtual memory, even if peak RSS is not large, and this breaks the fork call. I have no idea why. Running the tests outside of the Docker NixOS, it works completely fine, neither is virtual memory or RSS larger than 500 MiB.

Disabling that single test that uses Python multiprocessing with `disabledTestPaths = ["tests/test_BlockParallelReaders.py"];` works, but I fear that it will not pass review... I mean, I can also remove the test upstream so that I don't have to disable the test here ...

Actually, the tests in 0.6.2 run fine. And, by subsequently setting `rev = ` to various commits between v0.6.2 and v0.6.3, trying to build, copy-pasting the checksum shown in the failed built output, and trying to build again, I was able to find out that the tests broke with [this commit](https://github.com/mxmlnkn/ratarmount/commit/54381ad00c28ac4570d1977a3f157a2aaac2a710).

I then tried to push some changes to a new branch and test in the same manner by changing `rev =`, but suddenly I get an `illegal path references` error, and I have no idea why. I guess [this issue](https://github.com/NixOS/nix/issues/7150) is related, but I still don't know how to proceed from here. What I didn't yet try is to simply push to the master branch because I don't want to push untested commits there. But maybe it would fix the problem? That's the only remaining difference between this commit hash and the ones I tried to find the faulty commit.

My currently working derivation is:

```nix
{ lib
, buildPythonPackage
, fetchFromGitHub 
, pythonOlder
, pytestCheckHook
, indexed-bzip2
, indexed-gzip
, indexed-zstd
, python-xz
, setuptools
, rapidgzip
, rarfile
, zstandard     # Python bindings
, zstd          # System tool
}:

buildPythonPackage rec {
  pname = "ratarmountcore";
  version = "0.6.3";
  pyproject = true;

  disabled = pythonOlder "3.6";

  src = fetchFromGitHub {
    owner = "mxmlnkn";
    repo = "ratarmount";
    rev = "core-v${version}";
    hash = "sha256-2jG066BUkhyHRqRyFAucQRJrjXQNw2ccCxERKkltO3Y=";
    fetchSubmodules = true;
  };

  sourceRoot = "${src.name}/core";

  nativeBuildInputs = [ setuptools ];
  propagatedBuildInputs = [ indexed-gzip indexed-bzip2 indexed-zstd python-xz rapidgzip rarfile ];

  pythonImportsCheck = [ "ratarmountcore" ];

  nativeCheckInputs = [ pytestCheckHook zstandard zstd ];

  disabledTestPaths = [
    "tests/test_BlockParallelReaders.py"
  ];

  meta = with lib; {
    description = "Library for accessing archives by way of indexing";
    homepage = "https://github.com/mxmlnkn/ratarmount/tree/master/core";
    license = licenses.mit;
    maintainers = with lib.maintainers; [ mxmlnkn ];
  };
}
```

After changing the rev line to `rev = "23864e71d6861f56f1342a2031c749cf888d694a";` and removing the hash line, I get:

```
these 2 derivations will be built:
  /nix/store/9qpych3w4wjsglc1mqmp22mpbwphrx18-source.drv
  /nix/store/iyqfbak9jp1cxc9jm9zwiwq61pxrbd8s-python3.11-ratarmountcore-0.6.3.drv
building '/nix/store/9qpych3w4wjsglc1mqmp22mpbwphrx18-source.drv'...
exporting https://github.com/mxmlnkn/ratarmount.git (rev 23864e71d6861f56f1342a2031c749cf888d694a) into /nix/store/bqfq4db6nwycmkdrql9igsbrayqsw3g2-source
Initialized empty Git repository in /nix/store/bqfq4db6nwycmkdrql9igsbrayqsw3g2-source/.git/
remote: Enumerating objects: 642, done.
remote: Counting objects: 100% (642/642), done.
remote: Compressing objects: 100% (513/513), done.
remote: Total 642 (delta 88), reused 574 (delta 77), pack-reused 0
Receiving objects: 100% (642/642), 16.03 MiB | 7.24 MiB/s, done.
Resolving deltas: 100% (88/88), done.
From https://github.com/mxmlnkn/ratarmount
 * branch            fix-nix    -> FETCH_HEAD
 * [new branch]      fix-nix    -> origin/fix-nix
Switched to a new branch 'fetchgit'
removing `.git'...
error: illegal path references in fixed-output derivation '/nix/store/9qpych3w4wjsglc1mqmp22mpbwphrx18-source.drv'
error: 1 dependencies of derivation '/nix/store/iyqfbak9jp1cxc9jm9zwiwq61pxrbd8s-python3.11-ratarmountcore-0.6.3.drv' failed to build
```

`/nix/store/iyqfbak9jp1cxc9jm9zwiwq61pxrbd8s-python3.11-ratarmountcore-0.6.3.drv`

```python3
Derive(
    [("out"
      ,"/nix/store/bqfq4db6nwycmkdrql9igsbrayqsw3g2-source"
      ,"r:sha256"
      ,"0000000000000000000000000000000000000000000000000000000000000000")]
    ,[("/nix/store/2jizaab43mxxvdjxm8grkdvryshqwacg-stdenv-linux.drv",["out"])
      ,("/nix/store/33849ky0jdcmh081j24hack4gp3ssl5n-git-minimal-2.43.1.drv",["out"])
      ,("/nix/store/dynj352jjy6921i1kpbdq7bp7mymm5p3-bash-5.2p26.drv",["out"])
      ,("/nix/store/q0gsaxdmrlky0xz38q9i8zvdwxfsqyxy-nss-cacert-3.98.drv"
      ,["out"])]
    ,["/nix/store/17r5yk8sjmp2i96hyyv321vx1dz36dw6-nix-prefetch-git"
      ,"/nix/store/qkmrvp7kijcp8ky6isxkmxxbbsljhxmg-builder.sh"]
      ,"x86_64-linux"
      ,"/nix/store/087167dfxal194pm54cmcbbxsfy3cjgn-bash-5.2p26/bin/bash"
    ,["-e","/nix/store/qkmrvp7kijcp8ky6isxkmxxbbsljhxmg-builder.sh"]
    ,[("GIT_SSL_CAINFO","/nix/store/w5m8gk2nymbybqx37qqf97y2myxjci9j-nss-cacert-3.98/etc/ssl/certs/ca-bundle.crt")
      ,("__structuredAttrs","")
      ,("buildInputs","")
      ,("builder","/nix/store/087167dfxal194pm54cmcbbxsfy3cjgn-bash-5.2p26/bin/bash")
      ,("cmakeFlags","")
      ,("configureFlags","")
      ,("deepClone","")
      ,("depsBuildBuild","")
      ,("depsBuildBuildPropagated","")
      ,("depsBuildTarget","")
      ,("depsBuildTargetPropagated","")
      ,("depsHostHost","")
      ,("depsHostHostPropagated","")
      ,("depsTargetTarget","")
      ,("depsTargetTargetPropagated","")
      ,("doCheck","")
      ,("doInstallCheck","")
      ,("fetchLFS","")
      ,("fetchSubmodules","1")
      ,("fetcher","/nix/store/17r5yk8sjmp2i96hyyv321vx1dz36dw6-nix-prefetch-git")
      ,("impureEnvVars","http_proxy https_proxy ftp_proxy all_proxy no_proxy GIT_PROXY_COMMAND NIX_GIT_SSL_CAINFO SOCKS_SERVER")
      ,("leaveDotGit","")
      ,("mesonFlags","")
      ,("name","source")
      ,("nativeBuildInputs","/nix/store/59yz442mni4yy7b7kznmdim06k1ql52l-git-minimal-2.43.1")
      ,("nonConeMode","")
      ,("out","/nix/store/bqfq4db6nwycmkdrql9igsbrayqsw3g2-source")
      ,("outputHash","0000000000000000000000000000000000000000000000000000000000000000")
      ,("outputHashAlgo","sha256")
      ,("outputHashMode","recursive")
      ,("outputs","out")
      ,("patches","")
      ,("postFetch","")
      ,("preferLocalBuild","1")
      ,("propagatedBuildInputs","")
      ,("propagatedNativeBuildInputs","")
      ,("rev","23864e71d6861f56f1342a2031c749cf888d694a")
      ,("sparseCheckout","")
      ,("stdenv","/nix/store/8zzfhqpwghq0z8j4fwwdl5zy1aar0ap4-stdenv-linux")
      ,("strictDeps","")
      ,("system","x86_64-linux")
      ,("url","https://github.com/mxmlnkn/ratarmount.git")]
)
```

My Dockerfile for testing is:

```Dockerfile
FROM nixos/nix

RUN nix-channel --update

# Checkout the latest unstable branch to hopefully enable cached builds
# https://channels.nixos.org/nixpkgs-unstable/git-revision
RUN git clone --depth 1 https://github.com/NixOS/nixpkgs.git && cd nixpkgs && \
    git fetch --depth 1 origin bfa8b30043892dc2b660d403faa159bab7b65898 && \
    git checkout bfa8b30043892dc2b660d403faa159bab7b65898
# Set up Nixpkgs for cached builds
#RUN nix-env -iA cachix -f https://cachix.org/api/v1/install && cachix use nixpkgs
RUN nix-env -iA nixpkgs.vim nixpkgs.gnused nixpkgs.wget

CMD ["bash"]
```

I have no idea how to proceed from here.

**Comment by mxmlnkn** [comment]
Update:

I don't have the `os.fork` error on my home pc with 96 GB of RAM, only on my notebook with 16 GB of RAM. So, good to go with 0.6.3 ...?

**Comment by dotlambda** [comment]
> So, good to go with 0.6.3 ...?

I hope so. Please push your latest version and I'll test it.

**Comment by mxmlnkn** [comment]
I have pushed the 0.6.3 version.

My debugging adventure got even weirder. I was not able reproduce the `os.fork` error on the 16 GB system when using a local development build nix shell and running the test manually. Commands inside the NixOS Docker container:

```bash
git clone https://github.com/mxmlnkn/ratarmount.git
cd ratarmount/core
vim shell.nix
    with import <nixpkgs> {};
    with python311Packages;

    buildPythonPackage rec {
      name = "ratarmountcore";
      src = ".";
      dependencies = [ 
        indexed-gzip indexed-bzip2 indexed-zstd python-xz rapidgzip rarfile pytest mypy pylint zstandard zstd 
      ];
      propagatedBuildInputs = [ pkgs.libsndfile ];
    }
nix-env -iA nixpkgs.zstd
nix-shell shell.nix
    python3 -c 'import ratarmountcore as rmc; print(rmc.__version__)'  # 0.6.3
    sed -i 's|0[.]6[.]3|0.6.4|' ratarmountcore/version.py
    python3 -c 'import ratarmountcore as rmc; print(rmc.__version__)'  # 0.6.4
    python3 -m pytest tests/test_BlockParallelReaders.py
```
The last commands runs without problem and htop shows ~300MB of virtual memory and 50MB RSS. This is in contrast to the pytest called via `nix-build` command, which gobbles up 5-30GB of virtual memory and ~200MB RSS.

Screenshot_2024-03-06_22-42-34

This convinces me that it is not a problem solely on my side. There seem to be multiple issues (nix, Python multiprocessing, Docker?, ratarmountcore) colliding to produce this problem.

I opted for disabling the test to avoid issues on "low"-memory systems.

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 293224` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.ratarmount</li>
    <li>python312Packages.ratarmount.dist</li>
    <li>python312Packages.ratarmountcore</li>
    <li>python312Packages.ratarmountcore.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>ratarmount (python311Packages.ratarmount)</li>
    <li>ratarmount.dist (python311Packages.ratarmount.dist)</li>
    <li>python311Packages.ratarmountcore</li>
    <li>python311Packages.ratarmountcore.dist</li>
  </ul>
</details>

**Comment by mxmlnkn** [comment]
> * python312Packages.ratarmountcore

`indexed_zstd` did not built with Python 3.12 because it bundled the Cython sources instead of generating them. Should be fixed now.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 293224` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages built:</summary>
  <ul>
    <li>indexed-bzip2 (python311Packages.indexed-bzip2)</li>
    <li>indexed-bzip2.dist (python311Packages.indexed-bzip2.dist)</li>
    <li>python311Packages.indexed-gzip</li>
    <li>python311Packages.indexed-gzip.dist</li>
    <li>python311Packages.indexed-zstd</li>
    <li>python311Packages.indexed-zstd.dist</li>
    <li>rapidgzip (python311Packages.rapidgzip)</li>
    <li>rapidgzip.dist (python311Packages.rapidgzip.dist)</li>
    <li>ratarmount (python311Packages.ratarmount)</li>
    <li>ratarmount.dist (python311Packages.ratarmount.dist)</li>
    <li>python311Packages.ratarmountcore</li>
    <li>python311Packages.ratarmountcore.dist</li>
    <li>python312Packages.indexed-bzip2</li>
    <li>python312Packages.indexed-bzip2.dist</li>
    <li>python312Packages.indexed-gzip</li>
    <li>python312Packages.indexed-gzip.dist</li>
    <li>python312Packages.indexed-zstd</li>
    <li>python312Packages.indexed-zstd.dist</li>
    <li>python312Packages.rapidgzip</li>
    <li>python312Packages.rapidgzip.dist</li>
    <li>python312Packages.ratarmount</li>
    <li>python312Packages.ratarmount.dist</li>
    <li>python312Packages.ratarmountcore</li>
    <li>python312Packages.ratarmountcore.dist</li>
  </ul>
</details>

**Comment by wegank** [comment]
I can confirm that this PR has fixed the linked issue.

**Comment by wegank** [comment]
@ofborg build python311Packages.ratarmountcore python311Packages.ratarmountcore.passthru.tests

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 293224` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages built:</summary>
  <ul>
    <li>indexed-bzip2 (python311Packages.indexed-bzip2)</li>
    <li>indexed-bzip2.dist (python311Packages.indexed-bzip2.dist)</li>
    <li>python311Packages.indexed-gzip</li>
    <li>python311Packages.indexed-gzip.dist</li>
    <li>python311Packages.indexed-zstd</li>
    <li>python311Packages.indexed-zstd.dist</li>
    <li>rapidgzip (python311Packages.rapidgzip)</li>
    <li>rapidgzip.dist (python311Packages.rapidgzip.dist)</li>
    <li>ratarmount (python311Packages.ratarmount)</li>
    <li>ratarmount.dist (python311Packages.ratarmount.dist)</li>
    <li>python311Packages.ratarmountcore</li>
    <li>python311Packages.ratarmountcore.dist</li>
    <li>python312Packages.indexed-bzip2</li>
    <li>python312Packages.indexed-bzip2.dist</li>
    <li>python312Packages.indexed-gzip</li>
    <li>python312Packages.indexed-gzip.dist</li>
    <li>python312Packages.indexed-zstd</li>
    <li>python312Packages.indexed-zstd.dist</li>
    <li>python312Packages.rapidgzip</li>
    <li>python312Packages.rapidgzip.dist</li>
    <li>python312Packages.ratarmount</li>
    <li>python312Packages.ratarmount.dist</li>
    <li>python312Packages.ratarmountcore</li>
    <li>python312Packages.ratarmountcore.dist</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Pull request merged**